### PR TITLE
refactor: Decouple NetworkService in module and add interceptors

### LINKED ISSUE #180: Decouple NetworkService and add Picasso and Retrofit configs
Part of #174 

- Add Picasso and Retrofit configuration in module for Dagger Component
- Add Stetho network interceptor for both

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #180 

- Decouple NetworkService in modules
- Add network interceptors for Stetho in Retrofit and Picasso

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/common/di/module/NetworkModule.java [hunk 1]
```diff
@@ -1,23 +1,78 @@
...
+import retrofit2.converter.gson.GsonConverterFactory;
 
 @Module
 public class NetworkModule {
 
     @Provides
     @Singleton
-    EventService providesEventService() {
-        return NetworkService.getEventService();
+    ExclusionStrategy providesExcusionStartegy() {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/di/module/NetworkModule.java]: typo - `*Startegy()`

File: app/src/main/java/org/fossasia/openevent/app/OrgaApplication.java [hunk 2]
```diff
@@ -84,6 +87,10 @@ public void onCreate() {
         appComponent = DaggerAppComponent.builder()
             .androidModule(new AndroidModule(this))
             .build();
+
+        appComponent.inject(this);
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/OrgaApplication.java]: where was this injection before this PR?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/OrgaApplication.java]: We didn't need to inject picasso in the app, so nowhere

File: app/src/main/java/org/fossasia/openevent/app/common/di/module/NetworkModule.java (hunk 1 above)
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/di/module/NetworkModule.java]: Thanks

**Comment by iamareebjamal** [comment]
@arp95 @mahikaw @rohanagarwal94 @Shailesh351 please review

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge

**Pull request merged**